app/celery/tasks.py

```diff
@@ -113,7 +115,10 @@ def create_pdf_for_templated_letter(encrypted_letter_data):
     with current_app.test_request_context(''):
         html = HTML(string=str(template))
 
-    pdf = BytesIO(html.write_pdf())
+    try:
+        pdf = BytesIO(html.write_pdf())
+    except werkzeug.exceptions.BadGateway as exc:
```

Comment:
When this happened a few days ago I could only see a `weasyprint.urls.URLFetchingError` exception and not a `BadGateway` exception. How about just catching `Exception` here to cover anything going wrong?

Optional - a log line to tell us what type of exception was raised might then be useful

Reply:
The BadGateway is the top-level exception. The one you mentioned would be clearer, but it's not the one that we get, unfortunately. Here's an example stack trace:

```
...

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
...
  File "/usr/local/lib/python3.6/site-packages/weasyprint/urls.py", line 286, in fetch
    raise URLFetchingError('%s: %s' % (type(exc).__name__, str(exc)))
weasyprint.urls.URLFetchingError: URLError: <urlopen error _ssl.c:835: The handshake operation timed out>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
...
  File "/usr/local/lib/python3.6/site-packages/werkzeug/exceptions.py", line 807, in __call__
    raise self.mapping[code](*args, **kwargs)
werkzeug.exceptions.BadGateway: 502 Bad Gateway: The proxy server received an invalid response from an upstream server.
```

Looking again at the stack trace, I realised there was something strange going on. I've added another commit to explain and refactor the strangeness so it's a bit less strange.

Could you take a look?

Reply:
👍 that makes sense - I didn't spot the Bad Gateway error at first because I was looking at a different error log (the one for `Failed to load image at...` not for the task as a whole)